eng/testing/tests.mobile.targets

```diff
@@ -53,6 +62,17 @@
     <WriteLinesToFile File="$(PublishDir)xunit-excludes.txt" Lines="$(XunitExcludesTxtFileContent)" />
 
     <RemoveDir Directories="$(BundleDir)" />
+
+    <MonoAOTCompiler Condition="'$(RunAOTCompilation)' == 'true'"
+        CompilerBinaryPath="$(MicrosoftNetCoreAppRuntimePackNativeDir)cross\$(PackageRID)\mono-aot-cross"
+        Mode="Full"
+        OutputType="AsmOnly"
+        Assemblies="@(AotInputAssemblies)"
+        UseLLVM="$(MonoEnableLLVM)"
+        LLVMPath="$(MicrosoftNetCoreAppRuntimePackNativeDir)cross\$(PackageRID)">
+        <Output TaskParameter="CompiledAssemblies" ItemName="BundleAssemblies" />
```

Comment:
Are we using the `BundleAssemblies` item yet? Would it replace `SourceDir` which would require some tweaking to `AndroidAppBuilder`/`ApkBuilder`?

Reply:
Yeah, thanks, I noticed it as well. But firstly it should start to build in AOT w/o errors :)

Reply:
Mitchell, if you have any idea how to do it right, please let me know.

Reply:
I'm not entirely sure what all needs to be changed. I'm thinking that the `AndroidAppBuilderTask` takes in all the assemblies in `$(PublishDir)` as `SourceDir` which 
1) Checks that `$(MainLibraryFileName)` exists. 
2) zip all the files to `apk/assets/assets.zip` https://github.com/dotnet/runtime/blob/5761dd49339911241209c1ffd61105a94edb6df1/src/tasks/AndroidAppBuilder/ApkBuilder.cs#L106-L107
3) makes sure  libmonosgen-2.0.a exists
4) adds the `.so` files to the dynamic library

It seems like using the AOT'd assemblies `BundleAssemblies` would need to cover those steps instead.
@EgorBo would you happen to know if this is the right approach?

Reply:
Should be addressed now